pom.xml

```diff
@@ -31,7 +31,7 @@
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <!-- version properties -->
     <avro.version>1.7.7</avro.version>
-    <cdap.version>6.2.0</cdap.version>
+    <cdap.version>6.4.0-SNAPSHOT</cdap.version>
```

Comment:
Any specific reason why we are changing the CDAP version in here?

Reply:
Plugin has ETL Test, but tests are ignored if there are no Salesforce Account Credentials set in environment variable.  https://github.com/data-integrations/salesforce/blob/f3d38b2470f399a1869e7ad3129796b0dcadf73e/src/test/java/io/cdap/plugin/salesforce/etl/BaseSalesforceETLTest.java#L47-L60

With cdap.version 6.2.0 when etl test are executed we got this issue:

`2021-03-16 12:21:22,777 - WARN  [main:i.c.c.i.a.r.a.ArtifactInspector@252] - Plugin configuration type not supported. Plugin ignored. class io.cdap.plugin.salesforce.plugin.source.batch.SalesforceBatchSource
io.cdap.cdap.api.data.schema.UnsupportedTypeException: Only primitive and String types are supported`

which is related with:

`Plugins use the new macro substitution that supports object, which is introduced in 6.4.0.`

Reply:
I am still not clear why this change is needed? Isnt time type supported by cdap schema for 6.2 version?

Reply:
If you run the ETL Tests that are actually in the repo(
https://github.com/data-integrations/salesforce/blob/08b5980eec121c23a4529a5d5bcf805c5003776c/src/test/java/io/cdap/plugin/salesforce/etl/BaseSalesforceETLTest.java#L81-L89) you must specify Salesforce credentials here. 

Thus the ETL pipeline will fail with:

`2021-03-16 12:21:22,777 - WARN [main:i.c.c.i.a.r.a.ArtifactInspector@252] - Plugin configuration type not supported. Plugin ignored. class io.cdap.plugin.salesforce.plugin.source.batch.SalesforceBatchSource io.cdap.cdap.api.data.schema.UnsupportedTypeException: Only primitive and String types are supported`

By default, there are no credentials provided for Salesforce account and for this reason all the ETL Tests were skipped.

Please let us know if there's any other way we can run the ETL Tests in CDAP 6.4.0 without bumping the version.

Reply:
If you run the same test in 6.2 what happens then? the credentials are related to plugin. I am not sure why it matters what CDAP version we are using.

Reply:
When we run the same test in 6.2 we got this issue:

`2021-03-16 12:21:22,777 - WARN [main:i.c.c.i.a.r.a.ArtifactInspector@252] - Plugin configuration type not supported. Plugin ignored. class io.cdap.plugin.salesforce.plugin.source.batch.SalesforceBatchSource io.cdap.cdap.api.data.schema.UnsupportedTypeException: Only primitive and String types are supported`

we had same issue with PLUGIN-561 here is the discussion and solution https://github.com/data-integrations/salesforce/pull/65#discussion_r599082473

Reply:
Is this only needed for develop? This indicates plugin code from this branch will not work on cdap versions lower than 6.4?

Reply:
Maybe I can help clarify, nothing in this PR needs 6.4.0. The previous changes added in #61 added the 6.4.0 requirement. This PR is just updating the dependency so that tests can be run in the develop branch.

This code can be directly cherry-picked into the release/1.3 branch and we can leave the CDAP version as 6.2.0 there. But any future releases cut from develop will only work on 6.4.0+ due to the OAuth changes

Reply:
so the code for develop and release branch is diverged at this point. In that case, we should make sure the parent artifacts have corresponding cdap version so that plugins will not show up in any cdap versions lower than 6.4

Reply:
Yes, the parents on develop is already set for 6.4.0-SNAPSHOT, this was done as part of #61

Reply:
sounds good